ext/pg_result.c

```diff
@@ -1309,7 +1335,9 @@ yield_tuple(VALUE self, int ntuples, int nfields)
 	int tuple_num;
 	t_pg_result *this = pgresult_get_this(self);
 	VALUE result = pg_new_result(this->pgresult, this->connection);
+	t_pg_result *newthis = pgresult_get_this(result);
 	UNUSED(nfields);
+	newthis->flags = this->flags;
```

Comment:
I don't grok `pg_new_result`, so this is a naive question: should this copy happen inside `pg_new_result`?

Reply:
Hmm, good question! If I think about it a bit, I notice that the initial struct values are copied from `this->connection` in `pg_new_result`, but they should be copied from `this` instead. So currently `typemap=` is broken on `stream_each_tuple`, unless it's already set on the connection. I'll think a bit more about it, how this can be solved best.

Reply:
I fixed this issue in commit 6a450fae22ddbfd8a75545265e35a9143396870a on the master branch by introducing a result copy function instead of reusing `pg_new_result`. Thank you for highlighting this!